pkg/destroy/openstack/openstack.go

```diff
@@ -998,3 +1009,63 @@ func deleteImages(opts *clientconfig.ClientOpts, filter Filter, logger logrus.Fi
 	}
 	return true, nil
 }
+
+func untagRunner(opts *clientconfig.ClientOpts, networkTag string, logger logrus.FieldLogger) error {
+	backoffSettings := wait.Backoff{
+		Duration: time.Second * 10,
+		Steps:    25,
+	}
+
+	err := wait.ExponentialBackoff(backoffSettings, func() (bool, error) {
+		return untagNetwork(opts, networkTag, logger)
+	})
+	if err != nil {
+		return errors.Errorf("Unrecoverable error/timed out: %v", err)
```

Comment:
is "timed out" the only possible error here. If it is not it could make it difficult to debug.  'err' sould have enough info in its message.

Reply:
no, there could also be a bug with multiple networks with the same tag... I separated them as you ask.